building_sim_plugins/building_plugins_common/src/slotcar_common.cpp

```diff
@@ -186,11 +207,10 @@ void SlotcarCommon::path_request_cb(
 
     Eigen::Quaterniond quat(
       Eigen::AngleAxisd(msg->path[i].yaw, Eigen::Vector3d::UnitZ()));
-    trajectory[i].translation() = v3;
+    trajectory.at(i).translation() = v3;
     trajectory[i].linear() = Eigen::Matrix3d(quat);
```

Comment:
Since we are using `at` for indexing, for consistency it may be better if we use `at` throughout? I'm presuming we are using `at` here due to its safe access?

Reply:
> I'm presuming we are using at here due to its safe access?

Correct.

In order to reach line 211, the `trajectory.at(i)` of the previous line must have succeeded without throwing an exception, so line 211 is guaranteed to have an element with index `i`, making the bound check redundant.

That being said, the bound check of `.at(~)` is negligible overhead compared to everything else that's going on in the simulation, so I don't mind using it redundantly if others would prefer that.

Truthfully, neither of them should require an `.at(~)` since they were both explicitly resized earlier in this function, but I was experiencing some undefined behavior related to the trajectories, and so I put in this `.at(~)` to help with debugging it. I think the undefined behavior was related to multiple read/write issues, which should now be resolved by the new mutex, but I don't mind keeping the `.at(~)` here and adding one to line 211 just in case.